### PR TITLE
prevent hidden errors in buildspec

### PR DESCRIPTION
**Pull request by jesterhazy**
*Description of changes:*

prevent hidden errors in buildspec

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://sbpy3lu9z2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=8505f124-fe24-41fd-b655-a7e46194a5eb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: buildspec.yml [hunk 1]
```diff
@@ -19,16 +19,14 @@ phases:
...
+      - PREPROD_IMAGE="$ACCOUNT.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com/$ECR_REPO"
+      - PR_NUM=$(echo $CODEBUILD_SOURCE_VERSION | grep -o '[0-9]\+')
+      - echo "Pull request number: $PR_NUM. No value means this build is not from pull request."
       # keep ssh connection alive when communicating with remote ec2 server during integ test
       # largest connection idle time allowed: 10 seconds * 360 attempts = 60 minutes
-      - |
-        echo '  ServerAliveInterval 10' >> ~/.ssh/config
-        echo '  ServerAliveCountMax 360' >> ~/.ssh/config
+      -  echo '  ServerAliveInterval 10' >> ~/.ssh/config
+      -  echo '  ServerAliveCountMax 360' >> ~/.ssh/config
```
Comment (laurenyu) [inline buildspec.yml]: looks like there's an extra space after the dashes in these lines

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://sbpy3lu9z2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=63783c8d-4387-4d27-841b-acfaad24cd29%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://sbpy3lu9z2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=a22df016-0e37-4670-873d-25c647a4ba07%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**